addons/sms/static/tests/sms_widget_test.js

```diff
@@ -1,128 +1,202 @@
 /** @odoo-module **/
 
+import { addFakeModel } from '@bus/../tests/helpers/model_definitions_helpers';
+import { click, getFixture } from "@web/../tests/helpers/utils";
 import FormView from 'web.FormView';
 import ListView from 'web.ListView';
+import { setupViewRegistries } from "@web/../tests/views/helpers";
+import { start, startServer } from '@mail/../tests/helpers/test_utils';
 import testUtils from 'web.test_utils';
-import { makeView, setupViewRegistries } from "@web/../tests/views/helpers";
-import { getFixture } from "@web/../tests/helpers/utils";
 
 const createView = testUtils.createView;
 
-QUnit.module('fields', {
-    beforeEach: function () {
-        this.data = {
-            partner: {
-                fields: {
-                    message: {string: "message", type: "text"},
-                    foo: {string: "Foo", type: "char", default: "My little Foo Value"},
-                    mobile: {string: "mobile", type: "text"},
-                },
-                records: [{
-                    id: 1,
-                    message: "",
-                    foo: 'yop',
-                    mobile: "+32494444444",
-                }, {
-                    id: 2,
-                    message: "",
-                    foo: 'bayou',
-                }]
-            },
-            visitor: {
-                fields: {
-                    mobile: {string: "mobile", type: "text"},
-                },
-                records: [{
-                    id: 1,
-                    mobile: "+32494444444",
-                }]
-            },
+
+
+QUnit.module('SmsWidget', (hooks) => {
+
+    let target = undefined;
+
+    addFakeModel('fields.sms.emojis.partner', {
+        message: {string: "message", type: "text"},
+        foo: {string: "Foo", type: "char", default: "My little Foo Value"},
+        mobile: {string: "mobile", type: "text"},
+    });
+    addFakeModel('fields.sms.emojis.visitor', {
+        mobile: {string: "mobile", type: "text"},
+    });
+
+    /**
+     * Open the correct view based on inputs
+     *
+     * @param {string} model full name of the model to display
+     * @param {string} viewArch layout of the view in xml
+     * @param {number} newRecord true if the form should open a new record
+     * @param {function} mockRPC
+     * @param {boolean} debug
+     */
+    async function openTestView(model, viewArch, newRecord = true, mockRPC = null, debug = false) {
+        const views = {};
+        const resModel = model;
+        views[`${resModel},false,form`] = viewArch;
+
+        const pyEnv = await startServer();
+        const partnerRecord = pyEnv['fields.sms.emojis.partner'].create({ message: '', foo: 'yop', mobile: '+32494444444' });
+        const visitorRecord = pyEnv['fields.sms.emojis.visitor'].create({ mobile: '+32494444444' });
```

Comment:
Not sure the added value of having 2 separate models, as it makes the code more complicated.
I know you copied what was previously in place, but we could simplify if there is no need to have 2 models.